runtime/sema/checker.go

```diff
@@ -1862,6 +1950,80 @@ func (checker *Checker) checkCharacterLiteral(expression *ast.StringExpression)
 	)
 }
 
+func (checker *Checker) accessFromAstAccess(access ast.Access) (result Access) {
+
+	switch access := access.(type) {
+	case ast.PrimitiveAccess:
+		return PrimitiveAccess(access)
+	case ast.EntitlementAccess:
+
+		semaAccess, hasAccess := checker.Elaboration.GetSemanticAccess(access)
+		if hasAccess {
+			return semaAccess
+		}
+		defer func() {
+			checker.Elaboration.SetSemanticAccess(access, result)
+		}()
+
+		astEntitlements := access.EntitlementSet.Entitlements()
+		nominalType := checker.convertNominalType(astEntitlements[0])
+
+		switch nominalType := nominalType.(type) {
+		case *EntitlementType:
+			semanticEntitlements := make([]*EntitlementType, 0, len(astEntitlements))
+			semanticEntitlements = append(semanticEntitlements, nominalType)
+
+			for _, entitlement := range astEntitlements[1:] {
+				nominalType := checker.convertNominalType(entitlement)
+				entitlementType, ok := nominalType.(*EntitlementType)
+				if !ok {
+					// don't duplicate errors when the type here is invalid, as this will have triggered an error before
+					if nominalType != InvalidType {
+						checker.report(
+							&InvalidNonEntitlementAccessError{
+								Range: ast.NewRangeFromPositioned(checker.memoryGauge, entitlement),
+							},
+						)
+					}
+					result = PrimitiveAccess(ast.AccessNotSpecified)
+					return
+				}
+				semanticEntitlements = append(semanticEntitlements, entitlementType)
+			}
+			if access.EntitlementSet.Separator() == "," {
```

Comment:
Like @SupunS mentioned earlier, maybe make the separator kind an enumeration, and avoid the "stringly typing" and associated bugs it could caused